File: Data Structures & Algorithms/set-zeroes-in-matrix/submission-6.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        m, n = len(matrix), len(matrix[0])
        top = False
        for i in range(n):
            if matrix[0][i] == 0:
                top = True
                break
        for i in range(1, m):
            for j in range(n):
                if matrix[i][j] == 0:
                    matrix[0][j] = 0
                    matrix[i][0] = 0
        for i in range(1, m):
            for j in range(1, n):
                if matrix[i][0] == 0 or matrix[0][j] == 0:
                    matrix[i][j] = 0
        if matrix[0][0] == 0:
            for i in range(m):
                matrix[i][0] = 0
        if top:
            for i in range(n):
                matrix[0][i] = 0




        

        

File: Data Structures & Algorithms/car-fleet/submission-6.cpp
class Solution {
public:
    int carFleet(int target, vector<int>& position, vector<int>& speed) {
        vector<pair<int, double>> time;
        stack<double> fleet;
        for (int i = 0; i < position.size(); i++) {
            double t = speed[i] != 0 ? (double)(target - position[i]) / speed[i] : INFINITY;
            time.push_back({position[i], t});
        }
        sort(time.rbegin(), time.rend());
        for (const auto& [pos, t] : time) {
            if (fleet.empty()) {
                fleet.push(t);
            } else if (!fleet.empty() and fleet.top() < t) {
                fleet.push(t);
            }
        }
        return (int)fleet.size();
    }
};
